servant-client-core/src/Servant/Client/Core/Internal/Request.hs

```diff
@@ -56,8 +58,15 @@ data ServantError =
   -- | The content-type header is invalid
   | InvalidContentTypeHeader Response
   -- | There was a connection error, and no response was received
-  | ConnectionError Text
-  deriving (Eq, Show, Generic, Typeable)
+  | ConnectionError SomeException
+  deriving (Show, Generic, Typeable)
+
+instance Eq ServantError where
+  (FailureResponse r)                  == (FailureResponse r')            = r == r'
+  (DecodeFailure t r)                  == (DecodeFailure t' r')           = t == t' && r == r'
+  (UnsupportedContentType mt r)        == (UnsupportedContentType mt' r') = mt == mt' && r == r'
+  (InvalidContentTypeHeader r)         == (InvalidContentTypeHeader r')   = r == r'
+  (ConnectionError _)                  == (ConnectionError _)             = True
```

Comment:
You forgot non-equals cases:

```haskell
_ == _ = False
```

Reply:
Ha, I knew there was something missing. I rely too much on `-Wall` for these things.

Reply:
Fixed, thanks for the heads up